kong/conf_loader/init.lua

```diff
@@ -1339,19 +1339,35 @@ local function load(path, custom_conf, opts)
 
   conf = tablex.merge(conf, defaults) -- intersection (remove extraneous properties)
 
+  local is_nobody = false
+
   do
     -- nginx 'user' directive
     local user = utils.strip(conf.nginx_main_user):gsub("%s+", " ")
     if user == "nobody" or user == "nobody nobody" then
       conf.nginx_main_user = nil
+      is_nobody = true
     end
 
     local user = utils.strip(conf.nginx_user):gsub("%s+", " ")
     if user == "nobody" or user == "nobody nobody" then
       conf.nginx_user = nil
+      is_nobody = true
     end
   end
 
+  local kong_usr_exists = os.execute("getent passwd kong > /dev/null 2>&1")
+  local kong_grp_exists = os.execute("getent group kong > /dev/null 2>&1")
```

Comment:
will this work universally ie alpine, mac, windows?

Reply:
Mac 😢 
```
~$ getent
bash: getent: command not found
```

Maybe this is more universal? 🦆 

```suggestion
  local kong_grp_exists = os.execute("id -u kong > /dev/null 2>&1")
```

Any other (universal) ways to check if a specific user and group exist?

Reply:
This won't work universally.

`getent` is not POSIX. It won't run on Mac, Windows, Alpine.

But we don't create the `kong` user/group by default on these systems, i.e., this patch does not apply for Mac, Windows and Alpine: https://github.com/Kong/kong-build-tools/pull/339

So it will stay as `nobody` in this case.

Reply:
How about:

https://gist.github.com/dndx/b8b87f5abaac11ad89c89d8fd505a68f

This is POSIX and does not need any additional binary to do it.

Reply:
Great suggestion. Added you as a co-author and committed this change: https://github.com/Kong/kong/pull/6421/commits/95e913724e389a6f28d4a69f8c7dfcaec2c8c5f7